pkgs/build-support/rust/fetchcrate.nix

```diff
@@ -0,0 +1,35 @@
+{ lib, fetchurl, unzip }:
+
+{ crateName
+, version
+, sha256
+, ... } @ args:
+
+lib.overrideDerivation (fetchurl ({
```

Comment:
You likely want to use `overrideAttrs`.

Reply:
I don't know, I mostly copy-pasted from `fetchzip` and edited it. Why should I use `overrideAttrs` instead?

Reply:
`overrideDerivation` overrides the call to `derivation` whereas `overrideAttrs` overrides the call to `stdenv.mkDerivation`. Its typically relevant if something significant happens between `stdenv.mkDerivation` and `derivation`, which is often the case.

Reply:
Ok, the current code works, and the call to `overrideAttrs` would be slightly different I guess (and I'd have to guess how to use it by looking at its source code, because I can't find its documentation).

If `fetchzip` starts using `overrideAttrs`, I'll happily follow.